modules/react-components/src/components/modal/chunk-error-modal/chunk-error-modal.tsx

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { AppConstants } from "@wso2is/core/constants";
+import React, { FunctionComponent, ReactElement, useEffect, useState } from "react";
+import { Heading } from "../../typography";
+import { ConfirmationModal } from "../confirmation-modal";
+
+/**
+ * Prop type of the `ChunkErrorModal` component.
+ */
+interface ChunkModalPropTypes {
+    heading: string;
+    description: string;
+    primaryActionText: string;
+}
+
+/**
+ * This component listens to the `chunk load error` and pops up a modal
+ * when this error is occurred.
+ */
+export const ChunkErrorModal: FunctionComponent<ChunkModalPropTypes> = (
+    props: ChunkModalPropTypes
+): ReactElement => {
+    const { heading, description, primaryActionText } = props;
+
+    const [ showModal, setShowModal ] = useState(false);
+
+    /**
+     * Show modal.
+     */
+    const showErrorModal = () => {
```

Comment:
`useEffects` should be at the top for more readability. Since this has to be defined first shall we refactor the `useEffect` to straightaway `setShowModal(true)`. Then we can remove this function.